minor error: closing JabRef Logger

### JabRef version

Other (please describe below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 6.0--2024-11-05--4705977 Linux 6.12.6-200.fc41.x86_64 amd64  Java 21.0.5  JavaFX 23.0.1+4

### Checked with the latest development build (copy version output from About dialog)

- [x] I made a backup of my libraries before testing the latest development version.
- [x] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. I run JabRef from the shell (given https://github.com/JabRef/jabref/issues/11417)
2. upon shutting down JabRef, the last lines in the shell are an Error.

### Appendix

...
<details>

<summary>Log File</summary>

```
INFO: No open database detected
LOGGER ERROR: Failed to write log entry 'No open database detected' (java.lang.NullPointerException)
2025-01-13 17:33:24 [JabRef CachedThreadPool] org.jabref.logic.util.FallbackExceptionHandler.uncaughtException()
ERROR: Uncaught exception occurred in Thread[#99,JabRef CachedThreadPool,5,main]: org.apache.lucene.util.ThreadInterruptedException: java.lang.InterruptedException
	at org.apache.lucene.core@10.0.0/org.apache.lucene.index.IndexWriter.doWait(Unknown Source)
	at org.apache.lucene.core@10.0.0/org.apache.lucene.index.IndexWriter.forceMerge(Unknown Source)
	at org.jabref@6.0.6556/org.jabref.logic.search.indexing.DefaultLinkedFilesIndexer.optimizeIndex(Unknown Source)
	at org.jabref@6.0.6556/org.jabref.logic.search.indexing.DefaultLinkedFilesIndexer.closeIndex(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.InterruptedException
	at java.base/java.lang.Object.wait0(Native Method)
	at java.base/java.lang.Object.wait(Unknown Source)
	... 7 more
LOGGER ERROR: Failed to write log entry 'Uncaught exception occurred in Thread[#99,JabRef CachedThreadPool,5,main]' (java.lang.NullPointerException)

```

</details>

--- a/jablib/src/main/java/org/jabref/logic/search/sqlbased/indexing/DefaultLinkedFilesIndexer.java
+++ b/jablib/src/main/java/org/jabref/logic/search/sqlbased/indexing/DefaultLinkedFilesIndexer.java
@@ -43,6 +43,7 @@ import org.apache.lucene.search.TermQuery;
 import org.apache.lucene.search.TopDocs;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.util.ThreadInterruptedException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -305,6 +306,9 @@ public class DefaultLinkedFilesIndexer implements LuceneIndexer {
             indexWriter.forceMerge(1, true);
         } catch (IOException e) {
             LOGGER.warn("Could not force merge segments.", e);
+        } catch (ThreadInterruptedException e) {
+            LOGGER.warn("Interrupted optimization of index. ", e);
+            Thread.currentThread().interrupt();
         }
     }
 

--- a/jablib/src/main/java/org/jabref/logic/search/sqlbased/indexing/DefaultLinkedFilesIndexer.java
+++ b/jablib/src/main/java/org/jabref/logic/search/sqlbased/indexing/DefaultLinkedFilesIndexer.java
@@ -299,6 +299,9 @@ public class DefaultLinkedFilesIndexer implements LuceneIndexer {
                 indexWriter.forceMergeDeletes(true);
             } catch (IOException e) {
                 LOGGER.warn("Could not force merge deletes.", e);
+            } catch (ThreadInterruptedException e) {
+                LOGGER.debug("Interrupted optimization of index while forcing merge.");
+                Thread.currentThread().interrupt();
             }
         }
         try {
@@ -307,7 +310,7 @@ public class DefaultLinkedFilesIndexer implements LuceneIndexer {
         } catch (IOException e) {
             LOGGER.warn("Could not force merge segments.", e);
         } catch (ThreadInterruptedException e) {
-            LOGGER.warn("Interrupted optimization of index. ", e);
+            LOGGER.debug("Interrupted optimization of index.");
             Thread.currentThread().interrupt();
         }
     }

